### ISSUE TITLE
(Question)  Loading chunk failed. 

### ISSUE THREAD

**Issue by RogalDoe** [TiberiumFusion/FixedSteamFriendsUI#20]
Hello Comrade, 

I have a big question for older Client of Steam: 

Is this possible to manualy Fix that issue Loading chunk 7212 failed. 
(any Loading chunk (Number) failed. i searched to add manualy the missing part but i think thes request are intended to be inside package files ) but i have lake experience to proceed more 

Do you know what is causing this ? And is this Fixable ? 

Steam Client 

Version : May 30 203
Interfacing : v020

**Comment by RogalDoe** [comment]
And i Have a Big Crazy Project for Nostalgic people not related into the issue upper 


We will discuss that in a proper moment

**Comment by TiberiumFusion** [comment]
> Is this possible to manualy Fix that issue Loading chunk 7212 failed. (any Loading chunk (Number) failed. i searched to add manualy the missing part but i think thes request are intended to be inside package files ) but i have lake experience to proceed more
> 
> Do you know what is causing this ? And is this Fixable ?
> 
> Steam Client
> 
> Version : May 30 203 Interfacing : v020

What you are reporting is likely unrelated to FixedSteamFriendsUI.

The "loading chunk 1234 failed" message is due to incompatible changes Valve has made to their website network. Valve uses a hideous javascript bundler and loader to cobble their website together from many different pieces. When these pieces fail to run properly, the "loading chunk 1234 failed" message is created. For those of us using older Steam clients, the primary reason this happens is due to incompatible or invalid javascript from Valve. For example, the Valve website may try to control your VR headset (if you have one) by using the Steam client; older Steam clients lack this ability and the Valve website shits the bed in response instead of handling the problem. More recently, significant portions of the Valve website have become incompatible with the M86 capabilities of the older clients, which also results in the webpage failing to do anything properly. Lastly, the "loading chunk 1234 failed" message may appear if a certain file cannot be retrieved (i.e. it has been swiftly deleted by Valve as is typical for them), though this is unlikely in this scenario.

FixedSteamFriendsUI works around this problem by 1) using a snapshot of the steam-chat.com website which 2) includes all necessary javascript files and 3) conforms incompatible javascript to the M86 capabilities of our older Steam clients. You should not see any "loading chunk 1234 failed" messages in your friends list when using FixedSteamFriendsUI - afaik all necessary files have been identified and are present in the latest version (v4.0.0). If you see the "loading chunk 1234 failed" message in your friends list, then this may be a legitimate issue for FixedSteamFriendsUI to address (file missing from snapshot and/or it has incompatible/broken javascript).

Otherwise, if you are seeing "loading chunk 1234 failed" in the main Steam client window (*not* in the friends list), then you are experiencing an issue that cannot be fixed by FixedSteamFriendsUI. It is the problem with the entire Valve website network. I believe this is what you are reporting, since you did not specify precisely, and all prior reports of this type are regarding this. The only solution to that problem is to add missing capabilities to our older Steam clients or to fix the entire Valve website network to be compatible & compliant. Neither of these solutions are within scope of the FixedSteamFriendsUI project.

My workaround has been to use a normal web browser whenever I need to do something on a broken page of the Steam website. This is usually acceptable since I rarely ever want to use the horrible Steam website. It works well enough to log in, browse the store, buy things, view community hubs, etc. You can try that. The only other workaround I know of is to install a newer Steam client, which will fix the broken Valve website - but that of course defeats the purpose of not installing a newer Steam client.

> And i Have a Big Crazy Project for Nostalgic people not related into the issue upper
> 
> We will discuss that in a proper moment

This sounds unrelated and unfit to be discussed in detail here. Please keep issues on one single topic.

**Comment by RogalDoe** [comment]
> > Is this possible to manualy Fix that issue Loading chunk 7212 failed. (any Loading chunk (Number) failed. i searched to add manualy the missing part but i think thes request are intended to be inside package files ) but i have lake experience to proceed more
> > Do you know what is causing this ? And is this Fixable ?
> > Steam Client
> > Version : May 30 203 Interfacing : v020
> 
> What you are reporting is likely unrelated to FixedSteamFriendsUI.
> 
> The "loading chunk 1234 failed" message is due to incompatible changes Valve has made to their website network. Valve uses a hideous javascript bundler and loader to cobble their website together from many different pieces. When these pieces fail to run properly, the "loading chunk 1234 failed" message is created. For those of us using older Steam clients, the primary reason this happens is due to incompatible or invalid javascript from Valve. For example, the Valve website may try to control your VR headset (even if you don't have one) by using the Steam client; older Steam clients lack this ability and the Valve website shits the bed in response instead of handling the problem. More recently, significant portions of the Valve website have become incompatible with the M86 capabilities of the older clients, which also results in the webpage failing to do anything properly. Lastly, the "loading chunk 1234 failed" message may appear if a certain file cannot be retrieved (i.e. it has been swiftly deleted by Valve as is typical for them), though this is unlikely in this scenario.
> 
> FixedSteamFriendsUI works around this problem by 1) using a snapshot of the steam-chat.com website which 2) includes all necessary javascript files and 3) conforms incompatible javascript to the M86 capabilities of our older Steam clients. You should not see any "loading chunk 1234 failed" messages in your friends list when using FixedSteamFriendsUI - afaik all necessary files have been identified and are present in the latest version (v4.0.0). If you see the "loading chunk 1234 failed" message in your friends list, then this may be a legitimate issue for FixedSteamFriendsUI to address (file missing from snapshot and/or it has incompatible/broken javascript).
> 
> Otherwise, if you are seeing "loading chunk 1234 failed" in the main Steam client window (_not_ in the friends list), then you are experiencing an issue that cannot be fixed by FixedSteamFriendsUI. It is the problem with the entire Valve website network. I believe this is what you are reporting, since you did not specify precisely, and all prior reports of this type are regarding this. The only solution to that problem is to add missing capabilities to our older Steam clients or to fix the entire Valve website network to be compatible & compliant. Neither of these solutions are within scope of the FixedSteamFriendsUI project.
> 
> My workaround has been to use a normal web browser whenever I need to do something on a broken page of the Steam website. This is usually acceptable since I rarely ever want to use the horrible Steam website. It works well enough to log in, browse the store, buy things, view community hubs, etc. You can try that. The only other workaround I know of is to install a newer Steam client, which will fix the broken Valve website - but that of course defeats the purpose of not installing a newer Steam client.
> 
> > And i Have a Big Crazy Project for Nostalgic people not related into the issue upper
> > We will discuss that in a proper moment
> 
> This sounds unrelated and unfit to be discussed in detail here. Please keep issues on one single topic.

Daamnnn you answered that so great O_o i not expected that Accuracy a Big big thanks you for that !!! 


That means they adding or change some feature one theyr Web Site, and thes Old Steam Client Didn't have thes changement because they not supporting theyr older version by updating automaticaly theyr new Changement about Loading Chunk 


Its like we do not have the latest Chunk, but can we add it manualy into our Client because when you have a Loading Chunk Failled you have a precise Link with it like this one :

= From Store Game Page =
`Error Reference: Store_9427883_d48f6db3e8e96c8a
Loading chunk 7212 failed.
(missing: https://store.fastly.steamstatic.com/public/javascript/applications/store/chunk~115d6bfda.js?contenthash=b946a06869cda809657f)
`

```
test
```

You have that direct Link and i suppose that `chunk~115d6bfda.js` are the missing one we do not have in our client Right ? 


Im just here to discuss about it and learning if that possible to hotfix that Manualy because many many people asking it in Steam Forum and No one have the Resolve or the Method to Fix that 


In total we have maximum only 5 Loading Failled

And the Weird thing are this : your Patcher sems fixed 4 of them :|, the one you have on Shop Steam Point, on in Game Store ( You have 3 Loading Chunk Failed on a Game Store Page, 1 on Top - 2 at Middle ) Now only One on Top 



Keep in mind im here to understand, not to ask you fixing this, im just here to understand and ask question to a competent personn, because im find lack information about it 


And Thanks you alot for this Patcher i used only for testing other thing because i find something you use that regulate some Chunk on Steam and Steam Not Calling them again or maybe you added it into your patcher i don't know

**Comment by TiberiumFusion** [comment]
> Its like we do not have the latest Chunk, but can we add it manualy into our Client ?
> You have that direct Link and i suppose that chunk~115d6bfda.js are the missing one we do not have in our client Right ?

No. This is wrong.

Placing arbitrary Valve javascript "chunk" files from the Steam website into your Steam client's program files will achieve nothing, since 1) the Valve website never attempts to load itself from local files on your computer because 2) that is impossible.

There is no cross-origin javascript sharing between the Steam client and the Steam website. The Steam network has zero ability to access the javascript served by the Steam client's local http server. The Steam client has zero ability to access the javascript served by the Steam website's remote servers. This prevents any javascript-based javascript loader mechanism from loading javascript across an origin boundary.

> many many people asking it in Steam Forum and No one have the Resolve or the Method to Fix that

That is because it is a fool's errand, full of dangers and no future. The potential solutions are extremely expensive.

> And the Weird thing are this : your Patcher sems fixed 4 of them :|, the one you have on Shop Steam Point, on in Game Store ( You have 3 Loading Chunk Failed on a Game Store Page, 1 on Top - 2 at Middle ) Now only One on Top

It is difficult to understand what you are trying to say. My best guess is you are describing an experience similar to issue thread #12. You should read it.

Neither your explanation (as I assume to understand it) or the one given by the OP in #12 reproduces in my testing, so all I can do is speculate on it. Perhaps the inner frame in FriendsUI (steam-chat.com (or snapshot of it, using my patch)) loads other frames in the same origin context as itself _and_ the Steam client creates windows for them _and_ they reuse existing loaded javascript from the steam-chat.com webpage (or snapshot equivalent), of which the snapshot version includes my various fixes. This all seems highly unlikely though, and it doesn't properly explain it, but that's the kind of voodoo needed to explain it, if it's not simply typical user error or misunderstanding.

Either way, this is all out of scope of FixedSteamFriendsUI. If you are seeing some explainable side effect benefits when using FixedSteamFriendsUI, then that is all they are. Side effects. This is all I can tell you about them.

**Comment by RogalDoe** [comment]
> the Valve website never attempts to load itself from local files on your computer because 2) that is impossible.

aaaaa i see thes Line are Java Script Stored and loaded from Website but the Client Didn't have that referance then that why ? Impossible dammnnn :(, think we download the missing java chunk and reimplementing into steam but yeah (NOTHING IS IMPOSSIBLE JUST DO IT ) never mind just a Shia Leboeuf Phrase poped in my head 


But The client are suposed to load thes Chunk for the Website, why is not loading it ? :| 

Because The Client are just an Extention to load the Website Right ? 
Or
The Client are a Copy of the Web Site when they did that version and this same Version are locked in is Own Version ? and when you have Chunk Failled that because our Version are outdated of the Web Version ?

When you have a Chunk Failled is that mean Steam Web Server Traying to load a File supposed to be in Steam Client ? and unable to find it am i Right ?

**Comment by RogalDoe** [comment]
Also i tryed to load manualy with your Patch the missing chunk by creating the original Location and donwloaded the missing Js chunk and added this 2 line on index.html : 

    <script type="text/javascript">VALVE_PUBLIC_PATH = "https:\/\/store.fastly.steamstatic.com\/public\/";</script>
	<script type="text/javascript" src="public/javascript/applications/store/chunk~115d6bfda.js"></script>
	
	But not loaded and didn't disapear this shit : 
	`
	Error Reference: Store_9428986_d48f6db3e8e96c8a
Loading chunk 7212 failed.
(missing: https://store.fastly.steamstatic.com/public/javascript/applications/store/chunk~115d6bfda.js?contenthash=b946a06869cda809657f)
	`
	
	
	#Why not trying with my Noobs No Knowledge ah ah FOCK i tryed

**Comment by TiberiumFusion** [comment]
You are very confused. I already explained all of this - in detail, with specifics.

You are talking like someone who believes they can fill their car's fuel tank with water and drive a hundred miles on it. I have told you that is impossible, explained why, and yet you still believe you can drive your car with water in the fuel tank. I don't have time to give you a ten hour lecture about water, elementary school physics, and why water will not combust in a car engine. I already gave you the information specific to this scenario. It is your responsibility to learn your basic schooling about the thing you are seeking answers for.

You also obviously fail to realize that this project is expensive, I receive zero compensation for any of this, and I am a peasant. Personally teaching you the alphabet and ABCs is an outrageous waste of my very limited time.

tl;dr Tough love, welcome to the modern world. Try to find someone else who can afford to pay you the cost you are asking.
